tests/functional/s3-marker.sh

```diff
@@ -47,4 +47,74 @@ ${AWS} s3 rm s3://${BUCKET}/bb
 ${AWS} s3 rm s3://${BUCKET}/dd/dd
 ${AWS} s3 rm s3://${BUCKET}/ff
 
+
+# test marker with prefix on object
+
+for i in $(seq 1 21); do
+    ${AWS} s3 cp /etc/magic s3://${BUCKET}/subdir/object-$i
```

Comment:
I'm maybe quibbling, but I would create another subdir like `sub2dir` to verify that the objects of another directory are not shown.

Reply:
:+1: Will be done in next PR